File: Java/Eta/Applications/Shared/src/main/java/com/rtsdk/eta/shared/rdm/marketprice/MarketPriceUpdate.java
package com.rtsdk.eta.shared.rdm.marketprice;

import com.rtsdk.eta.codec.CodecFactory;
import com.rtsdk.eta.codec.DataTypes;
import com.rtsdk.eta.codec.Msg;
import com.rtsdk.eta.codec.MsgClasses;
import com.rtsdk.eta.codec.UpdateMsg;
import com.rtsdk.eta.rdm.DomainTypes;
import com.rtsdk.eta.rdm.InstrumentNameTypes;

/**
 * The market price update message. Used by an OMM Consumer, OMM Interactive provider and OMM Non-Interactive provider to
 * encode/decode a market price update message.
 */
public class MarketPriceUpdate extends MarketPriceResponseBase
{
    private UpdateMsg updateMsg = (UpdateMsg)CodecFactory.createMsg();

    public MarketPriceUpdate()
    {
        super();
    }

    /**
     * Clears the current contents of this object and prepares it for re-use.
     */
    public void clear()
    {
        super.clear();
    }
    
    
    @Override
    public Msg encodeMsg()
    {
        updateMsg.clear();
        updateMsg.msgClass(MsgClasses.UPDATE);
        updateMsg.streamId(streamId());
        updateMsg.domainType(DomainTypes.MARKET_PRICE);
        updateMsg.containerType(DataTypes.FIELD_LIST);
        
        if (checkHasServiceId())
        {
            /* attrib info */
            updateMsg.applyHasMsgKey();
            /* service Id */
            updateMsg.msgKey().serviceId(serviceId());
            updateMsg.msgKey().applyHasServiceId();
            /* name */
            updateMsg.msgKey().name(itemName());
            updateMsg.msgKey().applyHasName();
            /* name type */
            updateMsg.msgKey().nameType(InstrumentNameTypes.RIC);
            updateMsg.msgKey().applyHasNameType();
        }
        
        return updateMsg;
    }
    
    public String toString()
    {
        StringBuilder stringBuf = super.buildStringBuffer();
        stringBuf.insert(0, "MarketPriceUpdate: \n");
        return stringBuf.toString();
    }
    
    @Override
    public int domainType()
    {
        return DomainTypes.MARKET_PRICE;
    }
}

File: CSharp/Eta/Applications/Examples/EtaAppCommon/Rdm/item/marketprice/MarketPriceUpdate.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2022-2023 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.Codec;
using LSEG.Eta.Rdm;
using System.Text;

namespace LSEG.Eta.Example.Common
{
    public class MarketPriceUpdate : MarketPriceBase
    {
        private IUpdateMsg m_UpdateMsg = new Msg();

        public override int DomainType { get => (int)Rdm.DomainType.MARKET_PRICE; }
        public override int StreamId { get => m_UpdateMsg.StreamId; set { m_UpdateMsg.StreamId = value; } }
        public override int MsgClass { get => m_UpdateMsg.MsgClass; }

        public MarketPriceUpdate()
        {
            Clear();
        }
        
        public override Msg EncodeMsg()
        {   
            if (HasServiceId)
            {
                m_UpdateMsg.ApplyHasMsgKey();
                m_UpdateMsg.MsgKey.ApplyHasServiceId();
                m_UpdateMsg.MsgKey.ServiceId = ServiceId;

                m_UpdateMsg.MsgKey.Name = ItemName;
                m_UpdateMsg.MsgKey.ApplyHasName();
                m_UpdateMsg.MsgKey.NameType = InstrumentNameTypes.RIC;
                m_UpdateMsg.MsgKey.ApplyHasNameType();
            }

            return (Msg)m_UpdateMsg;
        }

        public override void Clear()
        {
            base.Clear();
            m_UpdateMsg.Clear();
            m_UpdateMsg.MsgClass = MsgClasses.UPDATE;
            m_UpdateMsg.DomainType = (int)Rdm.DomainType.MARKET_PRICE;
            m_UpdateMsg.ContainerType = DataTypes.FIELD_LIST;
        }

        public override string ToString()
        {
            StringBuilder stringBuf = PrepareStringBuilder();
            stringBuf.Insert(0, "MarketPriceUpdate: \n");
            return stringBuf.ToString();
        }
    }
}
